carrot-log4j-aws-sns/src/main/java/com/carrotgarden/log4j/aws/sns/Appender.java

```diff
@@ -203,38 +206,12 @@ public boolean hasAmazonClient() {
 
 	/** provide amazon login credentials from file */
 	protected boolean ensureCredentials() {
-
-        if (hasCredentials()) {
-            // Try to get URL and open stream
-            // Any exception, try as file
-            URL url = null;
-            try {
-                url = new URL(getCredentials());
-            } catch (MalformedURLException murle) {
-                url = Loader.getResource(getCredentials());
-            }
-
-            if (url != null) {
-                try {
-                    amazonCredentials = new PropertiesCredentials(url.openStream());
-                    return true;
-                } catch (Exception e) {
-                    // fail out
-                }
-            } else {
-                try {
-                    amazonCredentials = new PropertiesCredentials(new File(getCredentials()));
-                    return true;
-                } catch (Exception e2) {
-                    // fail out
-                }
-            }
-		}
-
-		LogLog.error("sns: invalid option", new IllegalArgumentException(
-				getCredentials() + " for Credentials"));
-
-		return false;
+		amazonCredentials = new AWSCredentialsProviderChain(new InstanceProfileCredentialsProvider(),
```

Comment:
Are you going to catch the possible exception here? I don't care, we can let it blow out the appender load. I can see benefits either way.